homeassistant/components/hydrawise/coordinator.py

```diff
@@ -31,15 +32,16 @@ class HydrawiseDataUpdateCoordinator(DataUpdateCoordinator[HydrawiseData]):
 
     api: Hydrawise
 
-    def __init__(
-        self, hass: HomeAssistant, api: Hydrawise, scan_interval: timedelta
-    ) -> None:
+    def __init__(self, hass: HomeAssistant, api: Hydrawise) -> None:
         """Initialize HydrawiseDataUpdateCoordinator."""
-        super().__init__(hass, LOGGER, name=DOMAIN, update_interval=scan_interval)
+        super().__init__(hass, LOGGER, name=DOMAIN, update_interval=MAIN_SCAN_INTERVAL)
         self.api = api
+        self._last_usage_refresh: datetime | None = None
 
     async def _async_update_data(self) -> HydrawiseData:
         """Fetch the latest data from Hydrawise."""
+        _utcnow = utcnow()
+        old_data = self.data
         # Don't fetch zones. We'll fetch them for each controller later.
         # This is to prevent 502 errors in some cases.
         # See: https://github.com/home-assistant/core/issues/120128
```

Comment:
While we are on it, can we combine `await self.api.get_user(fetch_zones=False)` and `await self.api.get_zones(controller)` by fetching the zones from the beginning? This would reduce two calls down to one.
In the old days, fetching zones required making separate API calls. But with the graphql API, we can fetch the zone data at the same time as fetching the user data.

Reply:
We can't combine the two because the GraphQL API returns 502 errors: https://github.com/home-assistant/core/issues/120128